dragonchain/job_processor/job_processor.py

```diff
@@ -55,6 +58,14 @@ def start() -> None:
     _kube = kubernetes.client.BatchV1Api()
 
     _log.debug("Job processor ready!")
+
+    if redis.llen_sync(PENDING_TASK_KEY):
+        _log.warning("WARNING! Pending job processor queue was not empty. Last job probably crashed. Re-queueing these dropped items.")
+        to_recover = redis.lrange_sync(PENDING_TASK_KEY, 0, -1, decode=False)
+        p = redis.pipeline_sync()
+        p.rpush(CONTRACT_TASK_KEY, *to_recover)
+        p.delete(PENDING_TASK_KEY)
```

Comment:
this is a race condition if something is added to `PENDING_TASK_KEY` after you check its length, but before you delete it here

Reply:
this is copied from queue.py

Reply:
meh, race condition should actually never happen as long as there is never more than 1 job processor running (which there shouldn't be), so I'll ignore this